src/screens/Home/index.tsx

```diff
@@ -87,4 +89,12 @@ class Home extends Component {
   }
 }
 
-export default Home;
+const mapStateToProps = (state) => ({
+  isAuthenticated: state.authenticate.isAuthenticated,
+});
+
+const mapDispatchToProps = dispatch => ({
+  userHasAuthenticated: (bool) => dispatch(authenticate(bool)),
```

Comment:
This one can be replaced by a HOC as well

Reply:
Updated